### PR TITLE
Clarify path_hierarchy documentation

### LINKED ISSUE #44709: `path_hierarchy` generates more tokens in reverse mode
(note: I reproduced this on the docker image for ES 7.2.0)

**Elasticsearch version** (`bin/elasticsearch --version`): `7.2.0`

**Plugins installed**: None.

**JVM version** (`java -version`): es reports jdk version `12.0.1`

**OS version**: `Linux 308bcfaae31e 4.9.125-linuxkit #1 SMP Fri Sep 7 08:20:28 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`

**Description of the problem including expected versus actual behavior**:

When I create a custom analyzer that uses the `path_hierarchy` tokenizer, and set the `reverse` option to `true`, I get more tokens than if reverse was not set. This happens for "invalid data".

According to the docs, the tokens should be in a reverse order - which they are ... but there are more tokens. From looking at it, I also note that:

* It seems as if the tokens from reverse mode, actually interpret the Hierarchy from Right-to-left, and thus isn't a simple reversing of the Left-to-right tokens ?
* This also happens with "valid data", `/parent/file`, this seems to confirm for me that the docs are wrong.

**Steps to reproduce**:
```
DELETE /bug_44709

PUT /bug_44709
{
    "settings": {
        "analysis": {
            "tokenizer": {
                "reverse_hierarchy": {
                    "type": "path_hierarchy",
                    "reverse": true
                }
            }
        }
    }
}

// This analysis returns 3 tokens
// /parent, /parent/, /parent//file
POST /bug_44709/_analyze
{
    "tokenizer": "path_hierarchy",
    "text": "/parent//file"
}

// This analysis returns 4 tokens
// /parent//file, parent//file, /file, file
POST /bug_44709/_analyze
{
    "tokenizer": "reverse_hierarchy",
    "text": "/parent//file"
}
```

**Provide logs (if relevant)**:

### PR DESCRIPTION
**Pull request by mayya-sharipova**
Closes #44709

### CODE REVIEW

File: docs/reference/analysis/tokenizers/pathhierarchy-tokenizer.asciidoc [hunk 1]
```diff
@@ -3,7 +3,9 @@
 
 The `path_hierarchy` tokenizer takes a hierarchical value like a filesystem
 path, splits on the path separator, and emits a term for each component in the
-tree.
+tree. `path_hierarcy` tokenizer uses Lucene's
```
Comment (jrodewig) [inline docs/reference/analysis/tokenizers/pathhierarchy-tokenizer.asciidoc]:
```suggestion
tree. The `path_hierarcy` tokenizer uses Lucene's
```

File: docs/reference/analysis/tokenizers/pathhierarchy-tokenizer.asciidoc [hunk 2]
```diff
@@ -3,7 +3,9 @@
...
+https://lucene.apache.org/core/8_0_0/analyzers-common/org/apache/lucene/analysis/path/PathHierarchyTokenizer.html[PathHierarchyTokenizer]
```
Comment (jrodewig) [inline docs/reference/analysis/tokenizers/pathhierarchy-tokenizer.asciidoc]: Not a big deal, but the `{lucene_version_path}` attribute ensures we use the appropriate Lucene version. 
```suggestion
https://lucene.apache.org/core/{lucene_version_path}/analyzers-common/org/apache/lucene/analysis/path/PathHierarchyTokenizer.html[PathHierarchyTokenizer]
```

File: docs/reference/analysis/tokenizers/pathhierarchy-tokenizer.asciidoc [hunk 3]
```diff
@@ -80,7 +82,9 @@ The `path_hierarchy` tokenizer accepts the following parameters:
     text has been consumed.  It is advisable not to change this setting.
 
 `reverse`::
-    If set to `true`, emits the tokens in reverse order.  Defaults to `false`.
+    If set to `true`, uses Lucene's
```
Comment (jrodewig) [inline docs/reference/analysis/tokenizers/pathhierarchy-tokenizer.asciidoc]:
```suggestion
    If `true`, uses Lucene's
```

File: docs/reference/analysis/tokenizers/pathhierarchy-tokenizer.asciidoc [hunk 4]
```diff
@@ -80,7 +82,9 @@ The `path_hierarchy` tokenizer accepts the following parameters:
...
+    http://lucene.apache.org/core/8_0_0/analyzers-common/org/apache/lucene/analysis/path/ReversePathHierarchyTokenizer.html[ReversePathHierarchyTokenizer],
```
Comment (jrodewig) [inline docs/reference/analysis/tokenizers/pathhierarchy-tokenizer.asciidoc]:
```suggestion
    http://lucene.apache.org/core/{lucene_version_path}/analyzers-common/org/apache/lucene/analysis/path/ReversePathHierarchyTokenizer.html[ReversePathHierarchyTokenizer],
```

**Review by jrodewig — APPROVED** [review]
Left a few non-blocking nits. Otherwise LGTM.

**Review by abdonpijpelink — APPROVED** [review]

**Pull request merged**